### ISSUE TITLE
The available()  should return Current!=null;

### ISSUE THREAD

**Issue by MikeTumapon** [thijse/Arduino-CmdMessenger#13]
For variable number of arguments, this code will not work:

    while (cmdMessenger.available())
    {
          int curNo=cmdMessenger.readInt16Arg();
          
          // Do something with this number.
    }

It will skip 1 parameter at a time.  Was this function intended for this?

**Comment by thijse** [comment]
Honestly, if forgett why I available goes to the next parameter. I do not see samples using it. I will think on your suggestion.

**Comment by Palatis** [comment]
I find this "feature" annoying, too...

**Comment by sigmaeo** [comment]
Me too.
Instead of:
`if( cmdMessenger.available() )
{
	value1 = cm.readInt32Arg();
	if( cmdMessenger.available() )
		value2 = cm.readInt32Arg();
}`

One have to do it this way:
`ui32 = cm.readInt32Arg();
if( cm.isArgOk() )
{
	value1 = ui32;
	ui32 = cm.readInt32Arg();
	if( cm.isArgOk() )
		ui32 = cm.readInt32Arg();
}`

If I send the command string (command-id=0): 
**0,1,2,3,4,5;**

The first example catches:
value1=2 and 
value2=4

The second example catches:
value1=1 and 
value2=2
as expected.

**Comment by thijse** [comment]
You are correct that this is a gotcha. available() is an alias for next(), which both *checks* and *advances* the argument cursor. So calling available() followed by a read*Arg() call advances twice — once in available() and once inside the read function — skipping every other argument.

The correct pattern for iterating a variable number of arguments is to call the read function directly and check isArgOk() afterwards:

# 1 <stdin>
# 1 <built-in>
# 1 <command-line>
# 1 <stdin>

Or simply call read*Arg() the exact number of times you expect arguments and check isArgOk() after each.

The available() / next() function is intended to be used to *peek* at whether more arguments exist before deciding which type to read — it is not safe to use it as a loop condition and then also call a read*Arg() function inside the loop body. This is a documentation gap; the API reference should make this clearer.

Closing as by-design behaviour with a documentation note. If you would like to propose renaming or splitting the function a pull request is welcome.
